inventorymanger/src/main/java/nl/hu/bep3/groep2/inventorymanger/core/application/InventoryCommandHandler.java

```diff
@@ -1,23 +1,53 @@
 package nl.hu.bep3.groep2.inventorymanger.core.application;
 
+import nl.hu.bep3.groep2.inventorymanger.core.application.command.MakeNewIngredients;
+import nl.hu.bep3.groep2.inventorymanger.core.application.command.OrderCreated;
+import nl.hu.bep3.groep2.inventorymanger.core.application.command.OrderUpdated;
+import nl.hu.bep3.groep2.inventorymanger.core.domain.Ingredient;
 import nl.hu.bep3.groep2.inventorymanger.core.ports.storage.InventoryRepository;
-import nl.hu.bep3.groep2.inventorymanger.infrastructure.driven.messaging.Publisher;
-import nl.hu.bep3.groep2.inventorymanger.infrastructure.driver.messaging.event.InventoryEvent;
 import org.springframework.stereotype.Service;
 
 @Service
 public class InventoryCommandHandler {
     private final InventoryRepository repository;
-    private final Publisher publisher;
 
-    public InventoryCommandHandler(InventoryRepository repository, Publisher publisher) {
+    public InventoryCommandHandler(InventoryRepository repository) {
         this.repository = repository;
-        this.publisher = publisher;
     }
 
-    public void handle(InventoryEvent inventoryEvent) {
-        inventoryEvent.getIngredients().keySet().forEach((name)->{
-            System.out.println(repository.findByName(name).orElseThrow(() -> new RuntimeException("doo doo brain")).getName());
+    public void handle(MakeNewIngredients command) {
+        command.getIngredients().keySet().forEach((name) -> {
+            Ingredient dbIngredient = repository.findByName(name)
+                    .orElseThrow(() -> new RuntimeException("ingredient not found"));
```

Comment:
Kun je hier aparte exceptions voor maken? Zodat we een IngredientNotFoundException hebben in plaats van RuntimeException